addons/purchase_stock/models/account_invoice.py

```diff
@@ -45,15 +45,16 @@ def _stock_account_prepare_anglo_saxon_in_lines_vals(self):
                 debit_expense_account = line._get_price_diff_account()
                 if not debit_expense_account:
                     continue
-                if line.product_id.cost_method != 'standard' and line.purchase_line_id:
 
-                    # Retrieve stock valuation moves.
-                    valuation_stock_moves = self.env['stock.move'].search([
-                        ('purchase_line_id', '=', line.purchase_line_id.id),
-                        ('state', '=', 'done'),
-                        ('product_qty', '!=', 0.0),
-                        ('product_id', '=', line.product_id.id),    # kits must be handled manually
-                    ])
+                # Retrieve stock valuation moves.
+                valuation_stock_moves = self.env['stock.move'].search([
```

Comment:
Could you just define an empty recordset and let the search in the if ? Sorry, I know we said to move the search outside the if-condition but I changed my mind: if the cost method is standard, the search is useless so let's save some time :)